examples/c/ex_col_store.c

```diff
@@ -0,0 +1,535 @@
+    error_check(cursor->close(cursor));
+}
+
+static void
+print_temp_column(WT_SESSION *session)
+{
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    uint8_t temp;
+
+    error_check(session->open_cursor(session, "colgroup:weather:temperature", NULL, NULL, &cursor));
+    printf("Temperature: \n");
+    while ((ret = cursor->next(cursor)) == 0) {
+        error_check(cursor->get_value(cursor, &temp));
+        printf("%d\n", temp);
+    }
+    scan_end_check(ret == WT_NOTFOUND);
+    error_check(cursor->close(cursor));
+}
+
+static void
+update_celsius_to_fahrenheit(WT_SESSION *session)
+{
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    uint8_t temp, temp_in_fahrenheit;
+
+    error_check(session->open_cursor(session, "colgroup:weather:temperature", NULL, NULL, &cursor));
+    while ((ret = cursor->next(cursor)) == 0) {
+        error_check(cursor->get_value(cursor, &temp));
+
+        /*
+         * Update the value from celsius to fahrenheit. Discarding decimals and keeping data simple
+         * by type casting to uint8_t.
+         */
+        temp_in_fahrenheit = (uint8_t)((1.8 * temp) + 32.0);
+
+        cursor->set_value(cursor, temp_in_fahrenheit);
+        error_check(cursor->update(cursor));
+    }
+    scan_end_check(ret == WT_NOTFOUND);
+    error_check(cursor->close(cursor));
+}
+
+static void
+chance_of_rain(WT_SESSION *session)
+{
+    WT_CURSOR *cursor;
+    uint64_t recno;
+    WT_DECL_RET;
+    uint16_t pressure;
+    uint8_t humidity;
+
+    error_check(
+      session->open_cursor(session, "colgroup:weather:humidity_pressure", NULL, NULL, &cursor));
+    while ((ret = cursor->next(cursor)) == 0) {
+        error_check(cursor->get_key(cursor, &recno));
+        error_check(cursor->get_value(cursor, &humidity, &pressure));
+        if (humidity > 70 && pressure < 1000)
+            printf("Rain likely\n");
+        else
+            printf("Rain unlikely\n");
+    }
+    scan_end_check(ret == WT_NOTFOUND);
+    error_check(cursor->close(cursor));
+}
+
+static void
+remove_country(WT_SESSION *session)
+{
+    WT_CURSOR *cursor;
+    uint64_t recno;
+    WT_DECL_RET;
+    uint16_t loc_lat, loc_long;
+    const char *country;
+
+    error_check(session->open_cursor(session, "colgroup:weather:location", NULL, NULL, &cursor));
+    /*
+     * All Australian data is being removed, to test if deletion works.
+     */
+    while ((ret = cursor->next(cursor)) == 0) {
+        error_check(cursor->get_key(cursor, &recno));
+        error_check(cursor->get_value(cursor, &loc_lat, &loc_long, &country));
+        if (strcmp("AUS", country) == 0) {
+            printf("Removing %s\n", country);
+            error_check(cursor->remove(cursor));
+        }
+    }
+    scan_end_check(ret == WT_NOTFOUND);
+    error_check(cursor->close(cursor));
+}
+
+static void
+generate_data(WEATHER *w_array)
+{
+    int country, day;
+
+    srand((unsigned int)getpid());
+
+    for (int i = 0; i < NUM_ENTRIES; i++) {
+        WEATHER w;
```

Comment:
In WT, we declare all the variables at the start of the function. Move `w` to the start of the function.